Gemfile

```diff
@@ -29,6 +29,11 @@ gem 'activerecord-import'
 
 group :development, :test do
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem 'guard', '~>2.16.2', require: false
+  gem 'guard-livereload', '~>2.5.2', require: false
+  gem 'guard-minitest', '~>2.4.6', require: false
+  gem 'rack-livereload'
+  gem 'foreman'
```

Comment:
@dtt101 Does foreman work in Docker?  I have never tried it as projects recently that have been using docker simply start up multiple containers (services). e.g webpacker is normally its own service in the docker-compose file but it runs the same image as the main app

Reply:
@dtt101 Do we not get hot-reloading with webpacker? I know all of our assets are not currently in packs, but it seems to be more common to moving more to be under the control of webpacker. And effectively removing everything out of the asset pipeline.

Reply:
Good questions (on guard, webpacker etc.)
I'll pick them all up here as they are kind of related.

1. I have used webpacker for react/js hot reloading but not for rails 'backend' code before, so part of the choice for Guard is that I have configured it before and know how it works. This may or may not be a good thing!
2. Previous to this PR the app currently, when run in docker, doesn't support any kind of hot reload (i.e when I change a html.erb file, I need to Ctrl + R to see the changes in the browser)
3. foreman does work in docker - used here because it felt like a nice way of encapsulating the commands

Reply:
I should add - totally happy to use webpacker if it is the thing!

Reply:
@dtt101 Thanks for the great reply. The explanation is sound. 

I was more curious as to why we might want guard. Like you I have more experience with it than the newer Rails features. 

I find it confusing that we - Rails developers - have webpacker but only want it for js and not all assets. But thats probably because most recent experiences have been backend focused and mainly API's. So all the FE stuff is in webpacker or whatever the trend is for JS/CSS management is (bower/webpack/etc).

The Asset pipeline was I think probably well ahead of its time.

Great to know about being able to use `foreman` in docker.

I'm trying to learn more about webpacker - so I can make the right decisions around if everything should move to it or not. Its still actively discouraged from the reading I do. Having said that not everybody agrees with DHH's or the Rails core teams thinking.

I personally like live-reload via Guard, but it did seem to me to go out of fashion.

Reply:
That's super interesting - thanks for sharing the thought process, that's super helpful.
If we want to merge this now, just to get the feature in, I am definitely happy to swap it out for webpacker down the line?

What do you reckon? It is all quite discreet and not much code.